coil-base/src/main/java/coil/memory/MemoryCache.kt

```diff
@@ -78,6 +77,43 @@ interface MemoryCache {
         override fun toString(): String {
             return "Key(key=$key, extras=$extras)"
         }
+
+        override fun writeToParcel(parcel: Parcel, flags: Int) {
+            parcel.writeString(key)
+            parcel.writeInt(extras.size)
+            extras.forEach {
+                parcel.writeString(it.key)
+                parcel.writeString(it.value)
+            }
+        }
+
+        override fun describeContents(): Int {
+            return 0
+        }
+
+        companion object {
+            @JvmField
+            val CREATOR: Parcelable.Creator<*> = Creator()
+        }
+
+        class Creator : Parcelable.Creator<Key> {
+            final override fun createFromParcel(parcel: Parcel): Key {
```

Comment:
the final keyword isn't needed per-se, and kotlin wants me to remove it, bit I couldn't find a way to make the API see "final" in there, and simply adding it works...

Reply:
Interesting - I think making a function non-final doesn't break binary compatibility so it's probably safe to remove. The class is also final by default.

Reply:
Feel free to remove them in a follow-up change if you're sure it's not breaking ( I'm leaning on that )
I tried to have a 1:1 match with current exposed API for simplicity.